rpc/doc.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Overview
+// Package rpc provides the underpinnings of requests and responses (server.Request)
+// and dispatching them to the API by defining a request
+// rpc provides the idea of a request that has a named type, a version, and a method to be called
+// and provides a mechanism – Find method – for matching a request to a callable function.
+
+// Overall Sequence Flow
+// 1. An API facade is registered with the root object's registry using a method like:
+//    registry.MustRegister("ModelManager", 10, ...)
+// 2. The RPC client sends the facade name and version number as part of the request to the RPC server.
+// 3. The RPC server resolves the facade name and version number, and executes the corresponding API function
+//    using conn.runRequest.
+
+// Key Components
```

Comment:
What follows here feels a bit too detailed, but maybe that's OK. At the same time, I wish there was something to help me connect the dots between this package and its subpackages, params and jsoncode.

Reply:
Sure, updated a componenet for this